lib/util/git/private-key.ts

```diff
@@ -5,14 +5,22 @@ import upath from 'upath';
 import { PLATFORM_GPG_FAILED } from '../../constants/error-messages';
 import { logger } from '../../logger';
 import { exec } from '../exec';
-import { newlineRegex } from '../regex';
+import type { ExecResult } from '../exec/types';
+import { newlineRegex, regEx } from '../regex';
 import { addSecretForSanitizing } from '../sanitize';
 
+type PrivateKeyFormat = 'gpg' | 'ssh';
+
+const sshKeyRegex = regEx(
+  /-----BEGIN ([A-Z ]+ )?PRIVATE KEY-----(.|\n)*?-----END ([A-Z]+ )?PRIVATE KEY-----/,
+);
```

Comment:
```suggestion
const sshKeyRegex = regEx(
  /-----BEGIN ([A-Z ]+ )?PRIVATE KEY-----.*?-----END ([A-Z]+ )?PRIVATE KEY-----/,
  's'
);
```

`s` causes `.` to match newlines 😉

Reply:
Nice! :ok_hand: Done: 275d0592bb583bc5144c11cc1d7b6c54d20fcaf5